x-pack/elastic-agent/pkg/core/monitoring/server/handler.go

```diff
@@ -27,15 +27,15 @@ type apiHandler struct {
 func (h *apiHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	err := h.innerFn(w, r)
 	if err != nil {
-		writeResponse(w, unexpectedErrorWithReason(err.Error()))
-
 		switch e := err.(type) {
 		case apiError:
 			w.WriteHeader(e.Status())
 		default:
 			w.WriteHeader(http.StatusInternalServerError)
 
 		}
+
+		writeResponse(w, unexpectedErrorWithReason(err.Error()))
```

Comment:
Possible to add a unit test for this code path? Having a unit test would have been an easy way to catch this.